dial_sync.go

```diff
@@ -96,13 +96,13 @@ func (ad *activeDial) start(ctx context.Context) {
 	ad.cancel()
 }
 
-func (ds *DialSync) getActiveDial(p peer.ID) *activeDial {
+func (ds *DialSync) getActiveDial(ctx context.Context, p peer.ID) *activeDial {
 	ds.dialsLk.Lock()
 	defer ds.dialsLk.Unlock()
 
 	actd, ok := ds.dials[p]
 	if !ok {
-		adctx, cancel := context.WithCancel(context.Background())
```

Comment:
This _needs_ to use the background context. Otherwise, if the first dialer cancels, we'll cancel the overall dial.

Reply:
This is also why the tests are failing.